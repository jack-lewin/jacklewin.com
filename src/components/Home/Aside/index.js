import React from 'react';

import AsideProps from '../../../prop-types/components/home/aside';
import styles from './styles.module.scss';

const Aside = ({ children }) => <aside className={styles.container}>{children}</aside>;

Aside.propTypes = AsideProps;

export default Aside;
