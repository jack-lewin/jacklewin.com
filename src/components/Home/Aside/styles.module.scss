@import '../../../mixins.scss';
@import '../../../functions.scss';

.container {
  display: flex;

  @include min-width(desktop) {
    flex-basis: 400px;
    max-width: 400px;
    padding-left: spacing(large);
  }
}
