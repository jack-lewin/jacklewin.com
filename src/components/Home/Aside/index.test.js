import React from 'react';
import { shallow } from 'enzyme';
import Aside from '.';

describe('Aside', () => {
  const props = {
    children: [<p key={1}>some</p>, <p key={2}>children</p>],
  };

  it('should render as expected', () => {
    const element = shallow(<Aside {...props} />);
    expect(element).toMatchSnapshot();
  });
});
