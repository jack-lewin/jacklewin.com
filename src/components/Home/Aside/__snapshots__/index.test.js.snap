// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Aside>
    <p>
      some
    </p>
    <p>
      children
    </p>
  </Aside>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          some
        </p>,
        <p>
          children
        </p>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "some",
        },
        "ref": null,
        "rendered": "some",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "children",
        },
        "ref": null,
        "rendered": "children",
        "type": "p",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            some
          </p>,
          <p>
            children
          </p>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "some",
          },
          "ref": null,
          "rendered": "some",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "children",
          },
          "ref": null,
          "rendered": "children",
          "type": "p",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
